src/mercat/mod.rs

```diff
@@ -573,111 +373,22 @@ pub struct InitializedTransferTxContent {
     pub auditors_payload: Vec<AuditorPayload>,
 }
 
-/// Wrapper for the initial transaction data and its signature.
-#[derive(Clone)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[cfg_attr(feature = "std", derive(Debug))]
-pub struct InitializedTransferTx {
-    pub content: InitializedTransferTxContent,
-    pub sig: Signature,
-}
-
-impl Encode for InitializedTransferTx {
-    #[inline]
-    fn size_hint(&self) -> usize {
-        self.content.size_hint() + schnorrkel::SIGNATURE_LENGTH
-    }
-
-    fn encode_to<W: Output>(&self, dest: &mut W) {
-        self.content.encode_to(dest);
-        self.sig.to_bytes().encode_to(dest);
-    }
-}
-
-impl Decode for InitializedTransferTx {
-    fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
-        let content = <InitializedTransferTxContent>::decode(input)?;
-        let sig = <[u8; schnorrkel::SIGNATURE_LENGTH]>::decode(input)?;
-        let sig = Signature::from_bytes(&sig)
-            .map_err(|_| CodecError::from("InitializedTx::sig is invalid"))?;
-
-        Ok(InitializedTransferTx { content, sig })
-    }
-}
-
 /// Holds the initial transaction data and the proof of equality of asset ids
 /// prepared by the receiver.
 #[derive(Clone, Encode, Decode)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub struct FinalizedTransferTxContent {
+pub struct FinalizedTransferTx {
     pub init_data: InitializedTransferTx,
     pub tx_id: u32,
     pub asset_id_from_sndr_equal_to_rcvr_proof: CipherEqualSamePubKeyProof,
 }
 
-/// Wrapper for the contents and the signature of the content sent by the
-/// receiver of the transaction.
-#[derive(Clone)]
-#[cfg_attr(feature = "std", derive(Debug))]
-pub struct FinalizedTransferTx {
-    pub content: FinalizedTransferTxContent,
-    pub sig: Signature,
-}
-
-impl Encode for FinalizedTransferTx {
-    #[inline]
-    fn size_hint(&self) -> usize {
-        self.content.size_hint() + schnorrkel::SIGNATURE_LENGTH
-    }
-
-    fn encode_to<W: Output>(&self, dest: &mut W) {
-        self.content.encode_to(dest);
-        self.sig.to_bytes().encode_to(dest);
-    }
-}
-
-impl Decode for FinalizedTransferTx {
-    fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
-        let content = <FinalizedTransferTxContent>::decode(input)?;
-        let sig = <[u8; schnorrkel::SIGNATURE_LENGTH]>::decode(input)?;
-        let sig = Signature::from_bytes(&sig)
-            .map_err(|_| CodecError::from("FinalizedTx::sig is invalid"))?;
-
-        Ok(FinalizedTransferTx { content, sig })
-    }
-}
-
-/// Wrapper for the contents, auditors' payload, and the signature of the justified and finalized
-/// transaction.
-#[derive(Clone)]
+/// Wrapper for the contents and auditors' payload.
+#[derive(Clone, Encode, Decode)]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub struct JustifiedTransferTx {
```

Comment:
we should just use `FinalizedTransferTx` in place of `JustifiedTransferTx`.

Reply:
Same answer as before